### PR TITLE
Animations for running/jumping

### PR DESCRIPTION
**Pull request by quigley-c**
Adds animations for player jumping, and updates the player running animation. Also changes the names of the NPCs to match their color and adds the animations to the NPCs and enemy.

Forked from #49

Addresses #3

### CODE REVIEW

File: src/objects/npc.lua [hunk 1]
```diff
@@ -30,14 +30,56 @@ return {
         self.is_walking = false
         self.direction = 'right'
 
+        self.colors = {
```
Comment (codeandkey) [inline src/objects/npc.lua]: The colors list here is a map for NPC names which is read on line 43 -- perhaps we can have separate fields for the NPC color and the NPC name in the map?

File: src/objects/player.lua [hunk 2]
```diff
@@ -30,13 +30,27 @@ return {
...
         self.idle = sprite.create('32x32_player.png', self.w, self.h, 0.25)
         self.walk = sprite.create('32x32_player-walk.png', self.w, self.h, 0.1)
+        self.jump = sprite.create('32x32_player-jump.png', self.w, self.h, 0.05)
+
+        self.jump.looping = false
 
         self.spr = self.walk
         self.spr:play()
 
+        self.spr = self.jump
```
Comment (codeandkey) [inline src/objects/player.lua]: I don't think these 2 are needed -- the sprite is assigned and started when the player jumps

File: src/objects/player.lua [hunk 3]
```diff
@@ -30,13 +30,27 @@ return {
...
+        self.spr:play()
+
         -- create a camera for the player
         self.camera = obj.create(self.__layer, 'camera', { x = self.x + self.w / 2, y = self.y + self.h / 2 })
     end,
+    explode = function(self, _, _)
+        -- turn the player into gibs
+        obj.create(self.__layer, 'gib', { img = '12x9_player_head.png', x = self.x + 11, y = self.y })
```
Comment (codeandkey) [inline src/objects/player.lua]: Might need update to newer 5-part gibs

**Comment by quigley-c** [comment]
Should be ready for merge. NPC coloring needs a rework anyway so I'll do that later.

**Pull request merged**